Topics/Software_Engineering/Agile.md

```diff
@@ -0,0 +1,98 @@
+
+## Four Agile Values
+
+1. **Individuals and interactions over processes and tools**:
+    - Agile teams value the collaboration and communication between team members. They prioritize face-to-face interactions and effective teamwork to ensure the success of the project. While processes and tools are important, they are secondary to having a good team and clear communication.
+    - **Example**: Consider a scenario where a complex algorithm needs to be implemented. In a process-centric approach, there might be a heavy reliance on tools and predefined procedures to execute the task. In contrast, an Agile team values the expertise and collaboration of its members. They may opt for collaborative coding sessions or discussions to find the most efficient solution. This focus on individual skills and teamwork fosters creativity and problem-solving, leading to more innovative and effective solutions.
+
+2. **Working software over comprehensive documentation**:
+    - Agile teams focus on delivering working software that provides value to the customer. While documentation is important, it is kept to a minimum and is only created when necessary. The main goal is to deliver a functioning product, not to produce extensive documentation.
+    - **Example**: In a traditional waterfall model, extensive documentation is often produced before any code is written. In an Agile project, the emphasis is on delivering working software at the earliest opportunity. For instance, instead of spending weeks creating detailed system documentation, the Agile team might prioritize delivering a minimum viable product (MVP) quickly. This allows stakeholders to interact with the actual software sooner, providing valuable insights and facilitating iterative improvements.
+
+3. **Customer collaboration over contract negotiation**:
+    - Agile teams prioritize regular and direct communication with the customer to understand their needs and expectations. This collaboration helps in delivering software that aligns with the customer's requirements. It's not just about delivering what's in the contract, but working together to ensure the product meets the customer's needs.
+    - **Example**: Imagine a software development team is working on an e-commerce platform. In a traditional, contract-centric approach, the team might strictly adhere to the initially agreed-upon features and functionalities. However, in an Agile environment, the team regularly collaborates with the customer to gather feedback. If the customer expresses a need for additional features like a personalized recommendation system, the Agile team can quickly adjust the development roadmap to incorporate these changes, ensuring the final product better meets the customer's vision.
```

Comment:
Good example! Can maybe include things like regularly presenting a demo of the produced software for the agile team, and for the contract team, if something is wrong with the final product because the customer changes their mind it could produce a lot of extra work due to some reason (explain)

Reply:
I see, I will try including what you suggested in my example!